middleware/tests/admin/test_sgx_authorization.py

```diff
@@ -0,0 +1,333 @@
+                SGXMigrationAuthorization.from_jsonfile("/an/existing/file.json")
+
+    def test_from_jsonfile_invalid_migration_spec(self):
+        # Test loading authorization with invalid migration spec
+        jsonsample = """
+        {
+          "version": 1,
+          "hashes": {
+            "exporter": "not-a-mrenclave",
+            "importer": "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+          },
+          "signatures": []
+        }
+        """
+
+        with patch("builtins.open", mock_open()) as open_mock:
+            open_mock.return_value.read.return_value = jsonsample
+
+            with self.assertRaises(ValueError):
+                SGXMigrationAuthorization.from_jsonfile("/an/existing/file.json")
+
+    def test_from_jsonfile_invalid_signatures(self):
+        # Test loading authorization with invalid signatures format
+        jsonsample = """
+        {
+          "version": 1,
+          "hashes": {
+            "exporter": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+            "importer": "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+          },
+          "signatures": "not-signatures"
+        }
+        """  # noqa E501
+
+        with patch("builtins.open", mock_open()) as open_mock:
+            open_mock.return_value.read.return_value = jsonsample
+
+            with self.assertRaises(ValueError):
+                SGXMigrationAuthorization.from_jsonfile("/an/existing/file.json")
+
+    def test_from_jsonfile_invalid_signature(self):
+        # Test loading authorization with invalid signature format
+        jsonsample = """
+        {
+          "version": 1,
+          "hashes": {
+            "exporter": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+            "importer": "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+          },
+          "signatures": [
+              "not-a-signature",
+              "304402201ef9d2a728e86aa3e8a0cf27a1f6afeba84af90f89ea50ea14483c4bd0c17fcd02201b6130ab0aed38128a4637b93ac90484aa2361c014e89c915d061fd27cab6aa6"
+          ]
+        }
+        """  # noqa E501
+
+        with patch("builtins.open", mock_open()) as open_mock:
+            open_mock.return_value.read.return_value = jsonsample
+
+            with self.assertRaises(ValueError):
+                SGXMigrationAuthorization.from_jsonfile("/an/existing/file.json")
+
+    def test_authorization_message(self):
+        # Test authorization message format
+        expected_msg = (b"\x19Ethereum Signed Message:\n" +
+                        b"160" +
+                        b"RSK_powHSM_SGX_upgrade_from_" +
+                        self.exporter_mrenclave.encode("ASCII") +
+                        b"_to_" + self.importer_mrenclave.encode("ASCII"))
+        self.assertEqual(expected_msg, self.sa.migration_spec.get_authorization_msg())
+
+    def test_authorization_digest(self):
```

Comment:
Isn't this a migration spec unit test?

Reply:
Nice catch, you're right, this is duplicated.

Reply:
This will be tackled in an upcoming PR